### DISCUSSION TITLE
cpg0011-lipocyteprofiler

### DISCUSSION THREAD

**Discussion by htdashti** [broadinstitute/cellpainting-gallery#65] [Dataset descriptions]
Imaging Platform project tag = 2018_04_12_T2D_V2F_Saadat_Broad AND 2022_04_14_Lipocyte_Profiler
Segmentation/ Feature extraction is being performed by Cimini lab
Profile creation is being performed by Cimini lab 
Data can be public in RODA Immediately

Update as generated:  

Link to "Batch1" and "Batch3" profile repo: https://github.com/broadinstitute/2018_04_12_T2D_V2F_Saadat_Broad
Link to other batches profile repo: ?
Link to publication: https://www.biorxiv.org/content/10.1101/2021.07.17.452050v1
cellpainting-gallery identifier: `cpg0011-lipocyteprofiler`  

- [ ] Metadata completely filled out in Project Profiler Database (Imaging Platform internal use only)
- [x] Segmentation/Feature extraction complete
- [x] Profiling complete

Transfer to CellPainting Gallery:
- [x] Upload data to RODA (is private by default)
- [ ] Run validation script to ensure completion
- [ ] Update cellpainting-gallery/README.md
- [ ] Make RODA entry public

If data is being published, prepare for publication:
- [ ] Run Distributed-BioFormats2Raw to create .ome.zarr files
- [ ] Upload (meta)data to IDR (images remain hosted in cellpainting-gallery).

Once published:
- [ ] Make IDR entry public
- [ ] Update cellpainting-gallery/README.md and open-data-registry/cellpainting-gallery.yml to reflect publication
- [x] Move this Issue from cellpainting-gallery-private to cellpainting-gallery. This step can be performed at an earlier point if it needs inputs from an external collaborator.

--- 

Hesam's original top post is below

Hi Shantanu,
We are organizing our LipocyteProfiler data according to the outline. We populated the branches based on our available data, and I was wondering if the following information are sufficient for the cellpainting-gallery? if yes, how can we upload the data, and if no, what are the required information?
<img width="423" alt="Screen Shot 2022-04-14 at 1 23 41 PM" src="https://user-images.githubusercontent.com/83827152/163441373-ab146722-24fa-4269-bf12-2f662357b276.png">
Thanks!

**Comment by shntnu** [comment]
@htdashti in #3 and #1, we are still figuring out the exact process here. But that's going to take a while so let's proceed with a few edits

- [x] @htdashti rename `LipocyteProfiler` to `cpg0010-lipoprofiler` or `cpg0011-lipocyteprofiler` (your call)
- [x] @htdashti rename `Broad-Inst` to `broad`
- [x] @htdashti point us to the location where we can access it on the Broad cluster and ensure we (any Broadie) can access it
- [x] @htdashti LMK if this should be made public right away, in which case @shntnu update `public_prefixes` in https://github.com/jump-cellpainting/cellpainting-gallery-config/blob/main/cdk.json to add `cpg0010-lipoprofiler` or `cpg0011-lipocyteprofiler` to the list and recreate the stack. 
- [x] @shntnu update `active_upload_prefixes ` in https://github.com/jump-cellpainting/cellpainting-gallery-config/blob/main/cdk.json to add `cpg0010-lipoprofiler` or `cpg0011-lipocyteprofiler` to the list and recreate the stack
- [x] @shntnu ask Beth to tag someone who has access to the Broad cluster **and** can assume `jump-cp-role`, to sync the data to the location using the command below _Instructions to sync_
- [ ] @htdashti Once I finish all steps, and assuming it is ok to make public right away, you are all set with the _Instructions to add to your README_ (below)

**Instructions to sync**

To actually copy the data, remove `--dryrun` after verifying that the dry runs look right

```sh
source=/web/ftp/incoming/hesam/ # replace this with the name of the folder that contains `cpg0011-lipocyteprofiler`
top_level=cpg0010-lipocyteprofiler

aws s3 sync \
  --dryrun \
  --profile jump-cp-role \
  --acl bucket-owner-full-control \
  --metadata-directive REPLACE  \
  ${source}/${top_level}/ \
  s3://cellpainting-gallery/${top_level}/  
```

**Instructions to add to your README**

The data are available on an S3 bucket.
They can be downloaded at no cost and no need for registration of any sort, using the command:

```bash
aws s3 sync \
  --no-sign-request \
  s3://cellpainting-gallery/cpg0011-lipocyteprofiler/ . 
```

AWS CLI installation instructions can be found [here](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html)

Note: If you'd like to just browse the data, it's a lot easier [to do so using a storage browser](https://stackoverflow.com/a/72143198/1094109).

**Comment by htdashti** [comment]
Thanks much @shntnu 
I applied the changes, and you can find the data under "/web/ftp/incoming/hesam/cpg0011-lipocyteprofiler/".
Yes, please make these public as soon as possible

**Comment by bethac07** [comment]
@htdashti 
```
-bash:uger-c018:~ 1001 $ cd /web/ftp/incoming/hesam/
-bash:uger-c018:/web/ftp/incoming/hesam 1002 $ ls
cpg0011-lipocyteprofiler/
-bash:uger-c018:/web/ftp/incoming/hesam 1003 $ cd cpg0011-lipocyteprofiler/
-bash: cd: cpg0011-lipocyteprofiler/: Permission denied
```

**Comment by htdashti** [comment]
@bethac07  Apologies! it should be fixed now.

**Comment by bethac07** [comment]
@htdashti The metadata csvs are missing; barcode_platemap.csv is the only thing in each metadata file. Please do add these so we can upload all in one shot, thanks!

**Comment by htdashti** [comment]
@bethac07 Thanks much! I think they should be in order now.

**Comment by shntnu** [comment]
> we are still figuring out the exact process here. But that's going to take a while so let's proceed with a few edits

I note that @ErinWeisbart has now created the process we will follow, which is for someone in the Cimini lab (in this case) to do this (private repo):

https://github.com/broadinstitute/cellpainting-gallery-private/issues/new?assignees=&labels=&template=data-immediately-public.md&title=YYYY_MM_DD_Dataset_Name

**Comment by bethac07** [comment]
@htdashti it looks like when you added the new files, somehow permissions were reset? 
```
-bash:login02:~ 1001 $ cd /web/ftp/incoming/hesam/cpg0011-lipocyteprofiler/
-bash:login02:/web/ftp/incoming/hesam/cpg0011-lipocyteprofiler 1007 $ ls broad/workspace/metadata/
ls: cannot open directory broad/workspace/metadata/: Permission denied
-bash:login02:/web/ftp/incoming/hesam/cpg0011-lipocyteprofiler 1008 $ cd broad/workspace/metadata/
-bash: cd: broad/workspace/metadata/: Permission denied
```

**Comment by htdashti** [comment]
Apologies! not sure whats happened, but I updated the permissions again.

**Comment by bethac07** [comment]
Upload underway!

**Comment by bethac07** [comment]
Sync completed over the long weekend, re-running now just to make sure nothing got missed, but I think we're all set!

**Comment by htdashti** [comment]
Great! Thank you very much!

**Comment by shntnu** [comment]
@htdashti 

Can you please add this to a future version of the [paper](https://www.biorxiv.org/content/10.1101/2021.07.17.452050v1) 

> All components of Cell Painting are available at the Cell Painting Gallery on the Registry of Open Data on AWS (https://registry.opendata.aws/cellpainting-gallery/) under accession number cpg0011.

**Comment by bethac07** [comment]
@htdashti Someone who read the preprint reached out to me asking for the pipelines - is it ok to share the ones for the batches present in this paper? It would be great if we can just add them all straight here in the gallery.

**Comment by htdashti** [comment]
@shntnu Thanks for the remainder. It is in the resubmitted materials.
@bethac07 Absolutely, thank you! It is a great suggestion, if you dont mind please add them to the gallery. please let me know if I can help with this

**Comment by ErinWeisbart** [comment]
@htdashti @bethac07 I've had someone ask me for the pipelines - it doesn't look like they ever made it into the gallery. I would also personally like to have the load_data.csv transferred to the gallery as well.

Looking at the metadata:
2019_10_29_Batch5 in the imaging platform has the plates that are in Batch4, Batch5, Batch6 in the gallery
2020_08_18_Batch6 in the imaging platform has the plates that are in Batch7, Batch8, Batch9 in the gallery
~I don't see the plates in Batch1 or Batch3 (gallery labels) in the imaging platform~ (Beth helped me track down Batch1 and Batch3 as part of a different project tag `2018_04_12_T2D_V2F_Saadat_Broad`)

I am happy to transfer the pipelines from 2019_10_29_Batch5 and 2020_08_18_Batch6 (imaging platform labels) as well as the load_data.csv that we have with your approval Hesam?

**Reply by ClaussnitzerLab** [reply]
Thanks! Yes, please go ahead with the transfer.

**Reply by bethac07** [reply]
Erin I know where the other pipelines likely are, can dig in later

**Reply by ErinWeisbart** [reply]
I diffed the illum.cppipe and analysis.cppipe in our originally named Batch4-6 and they were identical so I transferred them to CPG naming Batch4-9.

**Comment by ErinWeisbart** [comment]
@shntnu I see that you updated the initial comment to link to [this repo](https://github.com/broadinstitute/2019_08_03_Adipocyte_CellPainting_Claussnitzer/) as the publication repo and you edited the README in that repo to say that it has the data for cpg0011, but I don't understand how that is correct. That repo has 2021_11_19_Batch11 and later while the images in the CPG in cpg0011 are labeled Batch1-9 but map to our lab naming of 2019_10_29_Batch5 and 2020_08_18_Batch6.

Are there images/profiles that were published in Laber et al that did not make it into the CPG?
Is it possible that's not the correct data repo?

**Reply by ErinWeisbart** [reply]
I confirmed with Beth that [this repo](https://github.com/broadinstitute/2019_08_03_Adipocyte_CellPainting_Claussnitzer/) is NOT associated with cpg0011 but instead postdates it.

CPG labeled Batch1 plates are 2019_04_16_Batch1 in [this repo](https://github.com/broadinstitute/2018_04_12_T2D_V2F_Saadat_Broad). ~It is currently private but I have pinged there to see if we can make it public.~ It is now public.

I believe that CPG labeled Batch3 is 2019_08_06_Batch3 in that repo. The plate names in CPG were clearly made up post-hoc (`BR00000001`-`4`). The image names in the gallery indicate that:
BR00000001 = D14
BR00000002 = D0
BR00000003 = D3
BR00000004 = D8
If I compare the [platemaps in the GH repo](https://github.com/broadinstitute/2018_04_12_T2D_V2F_Saadat_Broad/tree/master/data/metadata/2019_08_06_Batch3/platemap) to the plate maps in the CPG using this mapping, they map on cell line (`subcutaneous` vs. `visceral`) and on patient (different identifiers used, but same pattern across wells).

```
BR = pd.read_csv('/Users/eweisbar/Desktop/platemap/BR00000003.tsv', sep='\t')
D = pd.read_csv('/Users/eweisbar/Downloads/Day3.txt', sep='\t')
D=D[['well_position','cell_line','patient']]
D.rename(columns={'cell_line':'cell_line_D'},inplace=True)
D['cell_line_D']=D['cell_line_D'].str.replace('sc','subcutaneous').replace('vc','visceral')
BR=BR[['well_position','Cell.Type','subj_ID']]
BR.rename(columns={'Cell.Type':'cell_line_BR'},inplace=True)
df = pd.merge(D,BR, on='well_position', how='inner')
df['cell_line_D'].equals(df['cell_line_BR'])
```

(I have not yet tracked down a repo for other batches)

**Comment by ErinWeisbart** [comment]
There's something missing - could be images, metadata, documentation, or some combination thereof.

All of the metadata files in the gallery have the same columns
```
platelist = []
for batch in [x for x in os.listdir('/Users/eweisbar/Desktop/platemaps') if 'Batch' in x]:
    for file in [x for x in os.listdir(f'/Users/eweisbar/Desktop/platemaps/{batch}/platemap/') if '.tsv' in x]:
        df = pd.read_csv(f'/Users/eweisbar/Desktop/platemaps/{batch}/platemap/{file}', sep='\t')
        platelist.append(file.split('.')[0])
        assert list(df.columns) == ['plate', 'Cell.Type', 'Day', 'row', 'col', 'batch', 'well_position','subj_ID']
```

None of those columns refers to a perturbation/drug. 

Figure 4 in Laber et al shows hepatocytes with and without isoproterenol, oleic acid, and metformin. The [README in the repo for the paper](https://github.com/ClaussnitzerLab/Lipocyte-Profiler/blob/main/README.md) says that the Fig 4b is in [generation of LP - isproterenol treatment.R](https://github.com/ClaussnitzerLab/Lipocyte-Profiler/blob/main/generation%20of%20LP%20-%20isoproterenol%20treatment.R) and Fig 4f+h is in [generation of LP in hepatocytes with drug treatment.R](https://github.com/ClaussnitzerLab/Lipocyte-Profiler/blob/main/generation%20of%20LP%20in%20hepatocytes%20%20with%20drug%20treatment.R)

In Fig 4b there are 7 plates listed - 3 are not in the CPG ["BR00112230", "BR00112232", "BR00112233"] and 4 are ['BR00112236', 'BR00112238', 'BR00112240', 'BR00112242']. Again, those 4 have the same metadata columns with no indication that they are or are not treated with isoproteranol.

In the Fig4f+h paper only two plates are explicitly named (BR00101180 and BR00101181) and neither are in the CPG.

**Reply by ErinWeisbart** [reply]
Hesam helped to track down perturbation information. 
- I have created a new Batch0 with additional images he sent. Batch0 has 4 plates: BR00112230, BR00112232, BR00112233 are from 2019_08_03_Adipocyte_CellPainting_Claussnitzer/2020_08_18_Batch6. I cannot find the original naming/source of BR00101181.
- Metadata csv's have been updated for ['BR00112236', 'BR00112238', 'BR00112240', 'BR00112242'] to include perturbation information. 
- BR00101180 was not used in the paper and so is not in the gallery.

**Comment by ErinWeisbart** [comment]
I've tracked down the load_data.csv files and am getting them ready for CPG. 
I started with updating the root structure and batch naming to match what's in the CPG.
```
import pandas as pd
import os
outdir = '/Users/eweisbar/Desktop/cpg0011_loadata_fixed'
root = '/Users/eweisbar/Desktop/forCPG_loaddata'
for batch in [x for x in os.listdir(root) if 'Batch' in x]:
    for plate in [x for x in os.listdir(os.path.join(root,batch)) if 'BR00' in x]:
        for file in ['load_data.csv','load_data_with_illum.csv']:
            df = pd.read_csv(os.path.join(root,batch,plate,file))
            # fix path root
            df.replace('projects/2018_04_12_T2D_V2F_Saadat_Broad','cpg0011-lipocyteprofiler/broad/images',regex=True,inplace=True)
            df.replace('projects/2019_08_03_Adipocyte_CellPainting_Claussnitzer','cpg0011-lipocyteprofiler/broad/images',regex=True,inplace=True)
            # fix batches
            df.replace('2019_04_16_Batch1',batch,regex=True,inplace=True)
            df.replace('2019_08_06_Batch3',batch,regex=True,inplace=True)
            df.replace('2019_09_30_Batch4',batch,regex=True,inplace=True)
            df.replace('2019_10_29_Batch5',batch,regex=True,inplace=True)
            df.replace('2020_08_18_Batch6',batch,regex=True,inplace=True)

            if not os.path.exists(os.path.join(outdir,batch,plate)):
                os.makedirs(os.path.join(outdir,batch,plate),exist_ok=True)
            df.to_csv(os.path.join(outdir,batch,plate,file), index=False)
assert len([x for x in os.listdir(root) if 'Batch' in x]) ==len([x for x in os.listdir(outdir) if 'Batch' in x])
for batch in [x for x in os.listdir(root) if 'Batch' in x]:
    assert len([x for x in os.listdir(os.path.join(root,batch)) if 'BR00' in x]) == len([x for x in os.listdir(os.path.join(outdir,batch)) if 'BR00' in x])
```


However, in spot checking things I've found that for many/all of the images in CPG:
1) plate folders didn't keep the original naming with the full plate name off the imager (e.g. BR00101077__2019-02-27T22_16_30-Measurement1) but instead were replaced with truncated plate names (e.g. BR00101077) images were uploaded directly to the plate
2) The `Images` folder that is below the plate folder when directly off the imager is not present
3) The Index.idx.xml file that contains the image metadata is not present
4) A large prefix has been appended to the file name (e.g. `subcutaneous__D14__4ca0419d-a567-4f62-b0c1-b66654b5a13b__BR00112243__r01c09f01p01-ch1sk1fk1fl1.tiff`) that is not present in the original load_data.csv (e.g. `r01c09f01p01-ch1sk1fk1fl1.tiff`)

So, rather than make significant edits on a plate-by-plate basis, I propose to upload the root and batch corrected .csv's to `load_data_csv_orig` (similar to as discussed [here](https://github.com/broadinstitute/cellpainting-gallery/issues/69)), track down and upload the `Index.idx.xml` files, and run [pe2loaddata](https://github.com/broadinstitute/pe2loaddata) on the files directly in CPG.

**Comment by ErinWeisbart** [comment]
None of the `illum` files were uploaded to the CPG though they were all made as part of analysis. We have copies of the illum's so I am unarchiving and transferring them from our end.

**Comment by ErinWeisbart** [comment]
Mapping of original naming/location to CPG naming/location:

| CPG Batch | Plate      | Original Project                               | Original Batch          | Notes                                     |
|-----------|------------|------------------------------------------------|-------------------------|-------------------------------------------|
| Batch1    | BR00101077 | 2018_04_12_T2D_V2F_Saadat_Broad                | 2019_04_16_Batch1 |                                           |
| Batch1    | BR00101079 | 2018_04_12_T2D_V2F_Saadat_Broad                |  2019_04_16_Batch1 |                                           |
| Batch1    | BR00101080 | 2018_04_12_T2D_V2F_Saadat_Broad                | 2019_04_16_Batch1 |                                           |
| Batch3    | BR00000001 | 2018_04_12_T2D_V2F_Saadat_Broad                | 2019_08_06_Batch3 | Plate originally named D14                |
| Batch3    | BR00000002 | 2018_04_12_T2D_V2F_Saadat_Broad                | 2019_08_06_Batch3 | Plate originally named Day0_AssayPlate0_1 |
| Batch3    | BR00000003 | 2018_04_12_T2D_V2F_Saadat_Broad                | 2019_08_06_Batch3 | Plate originally named Day3_AssayPlate0_1 |
| Batch3    | BR00000004 | 2018_04_12_T2D_V2F_Saadat_Broad                | 2019_08_06_Batch3 | Plate originally named Day8_AssayPlate0_3 |
| Batch4    | BR00101112 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2019_10_29_Batch5 |                                           |
| Batch4    | BR00101113 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer |  2019_10_29_Batch5 |                                           |
| Batch4    | BR00101114 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2019_10_29_Batch5 |                                           |
| Batch4    | BR00101149 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2019_10_29_Batch5 | Was also acquired in 2019_09_30_Batch4    |
| Batch5    | BR00101115 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer |  2019_10_29_Batch5 |                                           |
| Batch5    | BR00101116 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2019_10_29_Batch5 |                                           |
| Batch5    | BR00101117 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2019_10_29_Batch5 |                                           |
| Batch5    | BR00101118 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2019_10_29_Batch5 |                                           |
| Batch5    | BR00101146 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2019_10_29_Batch5 |                                           |
| Batch5    | BR00101147 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2019_10_29_Batch5 |                                           |
| Batch6    | BR00101139 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer |  2019_10_29_Batch5 |                                           |
| Batch6    | BR00101144 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2019_10_29_Batch5 |                                           |
| Batch6    | BR00101145 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2019_09_30_Batch4 | Was also acquired in 2019_10_29_Batch5    |
| Batch7    | BR00112226 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2020_08_18_Batch6 |                                           |
| Batch7    | BR00112227 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2020_08_18_Batch6 |                                           |
| Batch7    | BR00112240 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2020_08_18_Batch6 | Includes perturbations                    |
| Batch7    | BR00112241 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2020_08_18_Batch6 |                                           |
| Batch7    | BR00112242 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2020_08_18_Batch6 | Includes perturbations                    |
| Batch7    | BR00112243 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2020_08_18_Batch6 |                                           |
| Batch7    | BR00112244 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2020_08_18_Batch6 |                                           |
| Batch8    | BR00112228 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2020_08_18_Batch6 |                                           |
| Batch8    | BR00112234 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2020_08_18_Batch6 |                                           |
| Batch8    | BR00112235 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2020_08_18_Batch6 |                                           |
| Batch8    | BR00112236 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2020_08_18_Batch6 | Includes perturbations                    |
| Batch8    | BR00112237 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2020_08_18_Batch6 |                                           |
| Batch8    | BR00112238 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2020_08_18_Batch6 | Includes perturbations                   |
| Batch9    | BR00112231 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2020_08_18_Batch6 |                                           |
| Batch9    | BR00114011 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2020_08_18_Batch6 |                                           |
| Batch9    | BR00114012 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2020_08_18_Batch6 |                                           |
| Batch0    | BR00101181 | ?                                              | ?                       | Includes perturbations                    |
| Batch0    | BR00112230 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2020_08_18_Batch6 | Includes perturbations                    |
| Batch0    | BR00112232 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2020_08_18_Batch6 | Includes perturbations                    |
| Batch0    | BR00112233 | 2019_08_03_Adipocyte_CellPainting_Claussnitzer | 2020_08_18_Batch6 | Includes perturbations                    |
